src/test/e2e/api/admin/addon.e2e.test.ts

```diff
@@ -196,3 +202,98 @@ test('should not delete unknown addon configuration', async () => {
+    const payload = {
+        provider: 'htni',
+        enabled: true,
+        parameters: {},
+        events: [],
+    };
+
+    return app.request.post('/api/admin/addons').send(payload).expect(400);
+});
+
+test('updating an addon returns the new addon configuration', async () => {
+    const config = {
+        provider: 'webhook',
+        enabled: true,
+        parameters: {
+            url: 'http://localhost:4242/webhook',
+        },
+        events: [],
+    };
+    const { body } = await app.request.post('/api/admin/addons').send(config);
+
+    const updatedConfig = {
+        ...config,
+        enabled: false,
+        parameters: { url: 'http://new-url:4343' },
+    };
+
+    return app.request
+        .put(`/api/admin/addons/${body.id}`)
+        .send(updatedConfig)
+        .expect((res) => {
+            expect(res.body).toMatchObject(updatedConfig);
+        });
+});
+
+describe('missing descriptions', () => {
+    const addonWithoutDescription = {
+        provider: 'webhook',
+        enabled: true,
+        parameters: {
+            url: 'http://localhost:4242/webhook',
+        },
+        events: ['feature-created', 'feature-updated'],
+    };
+
+    test('creating an addon without a description, sets the description to `null`', async () => {
+        const { body } = await app.request
+            .post('/api/admin/addons')
+            .send(addonWithoutDescription)
+            .expect((res) => {
+                expect(res.body.description).toBeNull();
+            });
+
+        return app.request
+            .get(`/api/admin/addons/${body.id}`)
+            .expect((getResponse) => {
+                expect(getResponse.body.description).toBeNull();
+            });
+    });
+
+    test('updating an addon without touching `description` keeps the original value', async () => {
+        const { body } = await app.request
+            .post('/api/admin/addons')
+            .send(addonWithoutDescription);
+
+        const newUrl = 'http://localhost:4242/newUrl';
+        return app.request
+            .put(`/api/admin/addons/${body.id}`)
+            .send({ ...addonWithoutDescription, parameters: { url: newUrl } })
+            .expect((res) => {
+                expect(res.body.description).toBeNull();
+            });
+    });
+
+    test.each(['', null])(
+        'sending a description value of "%s", sets a `null` sets the description to an empty string',
+        async (description) => {
+            const { body } = await app.request
+                .post('/api/admin/addons')
+                .send(addonWithoutDescription);
+
+            return app.request
+                .put(`/api/admin/addons/${body.id}`)
+                .send({
+                    ...addonWithoutDescription,
+                    description,
+                })
+                .expect((res) => {
+                    expect(res.body.description).toStrictEqual('');
+                });
+        },
+    );
+});
```

Comment:
encoded how we deal with addon descriptions today. The last test (that it gets set to an empty string) is because of how the OpenAPI coerces a null into an empty string. Unsure what do about that here.